pynicotine/gtkgui/transferlist.py

```diff
@@ -327,31 +327,27 @@ def update_model(self, transfer=None, update_parent=True, forceupdate=False):
         if update_counters:
             self.update_num_users_files()
 
-        self.redraw_treeview()
+        self.tree_view.redraw()
 
     def update_parent_rows(self, transfer=None):
 
-        if self.tree_users != "ungrouped":
+        if self.grouping_mode != "ungrouped":
             if transfer is not None:
                 username = transfer.user
-                path = transfer.path if self.type == "download" else transfer.filename.rsplit("\\", 1)[0]
-                user_path = username + path
-
-                user_path_iter = self.paths.get(user_path)
-                user_iter = self.users.get(username)
+                user_path = username + self.get_transfer_path(transfer)
 
-                if user_path_iter:
-                    self.update_parent_row(user_path_iter, user_path, folder=True)
+                user_path_iter, user_path_child_transfers = self.paths[user_path]
```

Comment:
The path can be None for Queued/Incomplete items that are yet to start, leaving only the username without any folder. A KeyError can occur in Group By User grouping mode.